controller/dataplane/controller.go

```diff
@@ -176,6 +176,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	deploymentOpts := []k8sresources.DeploymentOpt{
 		labelSelectorFromDataPlaneStatusSelectorDeploymentOpt(dataplane),
 	}
+
+	log.Trace(logger, "ensuring generation of deployment configuration for KongPluginInstallations configured for DataPlane", dataplane)
+	kpisForDeployment, reconcileResult, err := ensureMappedConfigMapToKongPluginInstallationForDataPlane(ctx, logger, r.Client, dataplane)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("cannot ensure KongPluginInstallation for DataPlane: %w", err)
+	}
+	if reconcileResult.Requeue {
```

Comment:
I can see that for now `populateDedicatedConfigMapForKongPluginInstallation` nor `ensureMappedConfigMapToKongPluginInstallationForDataPlane` set `.RequeueAfter` but I'd consider adding a check for that here to be future proof and also requeue on that. WDYT?

Reply:
or maybe instead of returning `ctrl.Result` just return `requeue bool` and ensure returning configured `ctrl.Result{}` only in the body of `Reconcile`? WDYT?

Reply:
I'm ok with that as well but we already use `.RequeueAfter` in konnect entities reconciler: https://github.com/Kong/gateway-operator/blob/56f5cc1bfdfddfd92421b7efc84b5b78c40b55f5/controller/konnect/ops/ops.go#L141-L159.

That's not a big deal to use different mechanism between the 2 but let's just make sure to account for that field as well here.

Reply:
So here for Konnect it's handled OK. For the aforementioned, I think that approach with booleans is simpler, so I'll adjust it everywhere but in a separate PR